airflow/plugins/operators/airtable_to_warehouse.py

```diff
@@ -1,126 +1,192 @@
-            rename_fields (dict, optional): A mapping new desired column name (string)
-                to current airtable column name (string) respectively. Defaults to None.
-            column_prefix (str, optional): A string prefix to rename all columns with.
-                This prefix is not applied to columns affected by a rename triggered
-                from providing either `id_name` or `rename_fields`. Defaults to None.
+                Airtable Base
+            airtable_options (dict): optional fields: id_name, rename_fields,
+                column_prefix.
+                id_name (str, optional): The name to give the ID column. Defaults to "__id".
+                rename_fields (dict, optional): A mapping of raw column name from Airtable
+                    (string) to desired column name (string) respectively. Defaults to
+                    empty dict.
+                column_prefix (str, optional): A string prefix to rename all columns with.
+                    This prefix is not applied to columns affected by a rename triggered
+                    from providing either `id_name` or `rename_fields`.
             api_key (str, optional): The API key to use when downloading from airtable.
                 This can be someone's personal API key. If not provided, the environment
                 variable of `CALITP_AIRTABLE_API_KEY` is used.
         """
-        self.air_base_id = air_base_id
-        self.air_table_name = air_table_name
-        self.table_name = table_name
-        self.id_name = id_name
-        self.rename_fields = rename_fields
-        self.column_prefix = column_prefix
+        self.bucket = bucket
+        self.extract = AirtableExtract(
+            air_base_id=air_base_id,
+            air_base_name=air_base_name,
+            air_table_name=air_table_name,
+            **airtable_options,
+        )
         self.api_key = api_key
-        self.gcs_path = gcs_path
 
         super().__init__(**kwargs)
 
-    def execute(self, context):
-        df = airtable_to_df(
-            self.air_base_id,
-            self.air_table_name,
-            self.id_name,
-            self.rename_fields,
-            self.column_prefix,
-            self.api_key,
-        )
-
-        if self.table_name:
-            print(f"Writing table with shape: {df.shape}")
-            write_table(df, self.table_name)
-
-        if self.gcs_path:
-            clean_gcs_path = re.sub(r"\/+$", "", self.gcs_path)
-            gcs_file = (
-                f"{clean_gcs_path}/{context['execution_date']}/{self.table_name}.csv"
-            )
-            print(f"Uploading to gcs at {gcs_file}")
-            save_to_gcfs(df.to_csv(index=False).encode(), f"{gcs_file}", use_pipe=True)
+    def execute(self, **kwargs):
+        self.extract.fetch_from_airtable(self.api_key)
+        fs = get_fs()
+        self.extract.save_to_gcs(fs, self.bucket)
```

Comment:
I've kinda been wondering if bucket should be part of these pydantic models... mixed feelings.